File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.java
import java.util.*;
class Solution {
    public int solution(String[] babbling) {
        int answer = 0;

        StringBuilder sb = new StringBuilder();

        for(int i = 0; i < babbling.length; i++){
            String str = babbling[i];

            boolean aya = false;
            boolean ye = false;
            boolean woo = false;
            boolean ma = false;
            boolean check = false;
            int count = 0;

            for(int j = 0; j < str.length(); j++){
                char c = str.charAt(j);
                int index = 0;
                sb.append(c);

                if(sb.indexOf("aya") != -1){
                    if(aya) check = true;
                    index = sb.indexOf("aya");
                    sb.replace(index, index + 3, "-");
                    count++;
                    aya = true; ye = false; woo = false; ma = false;
                }else if(sb.indexOf("ye") != -1){
                    if(ye) check = true;
                    index = sb.indexOf("ye");
                    sb.replace(index, index + 2, "-");
                    count++;
                    aya = false; ye = true; woo = false; ma = false;
                }else if(sb.indexOf("woo") != -1){
                    if(woo) check = true;
                    index = sb.indexOf("woo");
                    sb.replace(index, index + 3, "-");
                    count++;
                    aya = false; ye = false; woo = true; ma = false;
                }else if(sb.indexOf("ma") != -1){
                    if(ma) check = true;
                    index = sb.indexOf("ma");
                    sb.replace(index, index + 2, "-");
                    count++;
                    aya = false; ye = false; woo = false; ma = true;
                }
                
                if(check) break;
            }

            if(!check && sb.length() == count) answer++;

            sb.setLength(0);
        }
        
        return answer;
    }
}

File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <string.h>

// babbling_len은 배열 babbling의 길이입니다.
// 파라미터로 주어지는 문자열은 const로 주어집니다. 변경하려면 문자열을 복사해서 사용하세요.
int solution(const char* babbling[], size_t babbling_len) {
    int answer = 0;
    int len = 0;
    int check = -1;
    
    const char* aya = "aya";
    const char* ye = "ye";
    const char* woo = "woo";
    const char* ma = "ma";
    
    for(int i = 0; i < babbling_len; i++){
        check = -1;
        len = strlen(babbling[i]);
        
        while(*babbling[i]){
            if(strncmp(babbling[i], aya, 3) == 0){
                if(check == 1) break;
                check = 1;
                babbling[i] += 3;
            }else if(strncmp(babbling[i], ye, 2) == 0){
                if(check == 2) break;
                check = 2;
                babbling[i] += 2;
            }else if(strncmp(babbling[i], woo, 3) == 0){
                if(check == 3) break;
                check = 3;
                babbling[i] += 3;
            }else if(strncmp(babbling[i], ma, 2) == 0){
                if(check == 4) break;
                check = 4;
                babbling[i] += 2;
            }else{
                break;
            }
        }
        
        if(*babbling[i] == '\0') answer++;
    }
    
    return answer;
}